src/lib/bootloader/mbr_update.rb

```diff
@@ -72,54 +72,26 @@ def install_generic_mbr
       end
     end
 
-    def set_parted_flag(disk, part_num, flag)
-      # we need at first clear this flag to avoid multiple flags (bnc#848609)
-      reset_flag(disk, flag)
-
-      # and then set it
-      command = ["/usr/sbin/parted", "-s", disk, "set", part_num, flag, "on"]
-      Yast::Execute.locally(*command)
-    end
-
-    def reset_flag(disk, flag)
-      command = ["/usr/sbin/parted", "-sm", disk, "print"]
-      out = Yast::Execute.locally(*command, stdout: :capture)
-
-      partitions = out.lines.select do |line|
-        values = line.split(":")
-        values[6] && values[6].match(/(?:\s|\A)#{flag}/)
-      end
-      partitions.map! { |line| line.split(":").first }
-
-      partitions.each do |part_num|
-        command = ["/usr/sbin/parted", "-s", disk, "set", part_num, flag, "off"]
-        Yast::Execute.locally(*command)
-      end
-    end
-
-    def can_activate_partition?(disk, num)
+    def can_activate_partition?(partition)
       # if primary partition on old DOS MBR table, GPT do not have such limit
-      gpt_disk = gpt?(disk)
+      gpt_disk = partition.disk.gpt?
 
-      !(Yast::Arch.ppc && gpt_disk) && (gpt_disk || num <= 4)
+      !(Yast::Arch.ppc && gpt_disk) && (gpt_disk || partition.is?(:primary, :extended))
     end
 
     def activate_partitions
-      partitions_to_activate.each do |m_activate|
-        num = m_activate["num"]
-        disk = m_activate["mbr"]
-        if num.nil? || disk.nil?
-          raise "INTERNAL ERROR: Data for partition to activate is invalid."
-        end
-
-        next unless can_activate_partition?(disk, num)
+      partitions_to_activate.each do |partition|
+        next unless can_activate_partition?(partition)
 
-        log.info "Activating partition #{num} on #{disk}"
+        log.info "Activating partition #{partition.inspect}"
         # set corresponding flag only bnc#930903
         if gpt?(disk)
-          set_parted_flag(disk, num, "legacy_boot")
+          # for legacy_boot storage_ng do not reset others, so lets
+          # do it manually
+          partition.siblings.select { |d| d.is?(:partition) }.each { |p| p.legacy_boot = false }
```

Comment:
yes, as gpt do not have primary versus logical and what I need is to reset other partitions, so I hope it will work, but of course I can change it to `partition.disk.partitions` instead